pkg/auth/gitcreds/gitcredentials_auth.go

```diff
@@ -0,0 +1,72 @@
+package gitcreds
+
+import (
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/jenkins-x/jx/pkg/auth"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+)
+
+// LoadGitCredentialsAuth loads the git credentials from the `git/credentials` file
+// in `$XDG_CONFIG_HOME/git/credentials` or in the `~/git/credentials` directory
+func LoadGitCredentialsAuth() (*auth.AuthConfig, error) {
```

Comment:
@jstrachan Have you seen https://github.com/jenkins-x/jx/pull/6034/files? We should abstract this behind a `ConfigHandler` interface which is used by the `ConfigService` to load the config.